prow/tide/status.go

```diff
@@ -507,3 +557,33 @@ func (sc *statusController) search() []PullRequest {
 func openPRsQuery(orgs, repos []string, orgExceptions map[string]sets.String) string {
 	return "is:pr state:open sort:updated-asc " + orgRepoQueryString(orgs, repos, orgExceptions)
 }
+
+const indexNamePassingJobs = "tide-passing-jobs"
+
+func indexKeyPassingJobs(org, repo, baseSHA, headSHA string) string {
+	return fmt.Sprintf("%s/%s@%s+%s", org, repo, baseSHA, headSHA)
+}
+
+func indexFuncPassingJobs(obj runtime.Object) []string {
+	pj, ok := obj.(*prowapi.ProwJob)
+	// Should never happen
```

Comment:
So according to the [godoc contract](https://github.com/kubernetes-sigs/controller-runtime/blob/47744b5a819ddb704a85e6b11408cdf45443861b/pkg/cache/informer_cache.go#L147) we are  guaranteed to get objects of the current type. I added it mainly because its obviously not enforced by the compiler, so if this ever happens to not be the case, we'll get a NPD here without the `ok`.

Do you prefer keeping or removing it?

Reply:
If this does happen then we should at least warn about it. Since it seems bizarre I'd be okay with panicing too, although warning is probably better.

Reply:
Yeah I've actually changed my opinion and changed it so that it panics. Should we ever get the wrong type here something went very wrong and we can not assume we'll get the correct result when querying data later on